k8s/superset/chart/superset/templates/postgresql-sts.yaml

```diff
@@ -0,0 +1,124 @@
+  serviceName: {{ .Release.Name }}-postgresql-service
+  template:
+    metadata:
+      labels: *PostgreSQLLabels
+    spec:
+      serviceAccountName: {{ .Release.Name }}-serviceaccount
+      volumes:
+      - name: tls-volume
+        secret:
+          secretName: {{ .Release.Name }}-tls
+      - name: {{ .Release.Name }}-postgresql-pvc
+        persistentVolumeClaim:
+          claimName: {{ .Release.Name }}-postgresql-pvc
+      containers:
+      - name: postgresql-server
+        image: {{ .Values.postgresql.image }}
+        imagePullPolicy: Always
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        ports:
+        - containerPort: 5432
+          name: postgres
+        livenessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          failureThreshold: 6
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 5
+          timeoutSeconds: 3
+          periodSeconds: 5
+        volumeMounts:
+        - name: {{ .Release.Name }}-postgresql-pvc
+          mountPath: /var/lib/postgresql/data
+        - name: tls-volume
+          mountPath: /var/lib/postgresql/data/tls
+        env:
+        - name: "PGDATA"
+          value: "/var/lib/postgresql/data/pgdata"
+        - name: "POSTGRES_PASSWORD"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-postgresql-secret
+              key: password
+        - name: "POSTGRES_USER"
+          value: {{ .Values.postgresql.db.user}}
+        - name: "POSTGRES_DB"
+          value: {{.Values.postgresql.db.name}}
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+      - name: postgresql-exporter
+        image: {{ .Values.postgresql.exporter.image }}
+        imagePullPolicy: Always
+        ports:
+          - containerPort: 9187
+            name: exporter
+        env:
+          - name: "DATA_SOURCE_URI"
+            value: "localhost:5432/postgres?sslmode=disable"
+          - name: "DATA_SOURCE_USER"
+            value: "postgres"
+          - name: "DATA_SOURCE_PASS"
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Release.Name }}-postgresql-secret
+                key: password
+      {{ if .Values.metrics.exporter.enabled }}
+      - name: prometheus-to-sd
+        image: {{ .Values.metrics.image }}
+        ports:
+          - name: profiler
+            containerPort: 6060
+        command:
+          - /monitor
+          - --stackdriver-prefix=custom.googleapis.com
+          - --source=postgresql:http://localhost:9187/metrics
+          - --pod-id=$(POD_NAME)
+          - --namespace-id=$(POD_NAMESPACE)
+          - --monitored-resource-types=k8s
```

Comment:
Have you tested it with the `prometheus-to-sd` enabled ? 
AFAIK you may need to use `--monitored-resource-type-prefix` instead of `--monitored-resource-types`.

Reply:
app/install worked with

metrics:
  image: null
  exporter:
    enabled: true